keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/CheckScheduler.kt

```diff
@@ -94,8 +95,12 @@ class CheckScheduler(
                    */
                   withTimeout(checkTimeout.toMillis()) {
                     launch {
-                      resourceActuator.checkResource(it)
-                      publisher.publishEvent(ResourceCheckCompleted(Duration.between(startTime, clock.instant())))
+                      try {
+                        resourceActuator.checkResource(it)
+                        publisher.publishEvent(ResourceCheckCompleted(Duration.between(startTime, clock.instant())))
+                      } catch (e: EnvironmentCurrentlyBeingActedOn) {
+                        log.warn("Couldn't actuate resource ${it.id} because environment is currently being acted on", e)
```

Comment:
Might be a nit, but I think words like "actuation" or "acted on" are kind of vague and not super user-friendly. If/when we expose a message to users in the UI, maybe we could say something like "...because resources in the environment are currently being created of modified"?

Reply:
That's a good idea, will do this in a future PR.